redux_userlist_06/src/components/Dashboard/getList.tsx

```diff
@@ -0,0 +1,60 @@
+import "./styles.css";
+import { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { addUserList } from "../../actions/actions";
+import { UserDetails } from "../../interfaces";
+import UserProfile from "./userProfile";
+import axios from "axios";
+import * as constants from "../../constants/constants";
+import { Link } from "react-router-dom";
+
+function GetList() {
+  const usersState: UserDetails[] = useSelector(
+    (state: any) => state.UserReducer
+  );
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (usersState.length === 0) {
```

Comment:
why do we need this condition?

Reply:
As we move back from update page to dashboard/getList , the useffect is called and data is fetched again from the api and added into store, to avoid this behaviour i added it. Do we have other solutions for this?  I tried to put api call in App.tsx, but not working.